burnman/seismic.py

```diff
@@ -222,45 +280,59 @@ def v_s(self, depth):
 
         return self._lookup(depth, self.table_vs)
 
+    def QK(self, depth):
+
+        return self._lookup(depth, self.table_Qk)
+
+    def QG(self, depth):
+
+        return self._lookup(depth, self.table_Qmu)
+
 
     def density(self, depth):
 
         return self._lookup(depth, self.table_density)
 
     def depth(self, pressure):
-        if pressure > max(self.table_pressure) or pressure < min(self.table_pressure)  :
-           raise ValueError, "Pressure outside range of PREM"
-        radius = np.interp(pressure, self.table_pressure[::-1], self.table_radius[::-1] )
-        return self.earth_radius - radius
+        if self.table_depth[0]:
```

Comment:
Well, rebasing to master got me a new git trauma...

And test_partitioning.py is failing. I'm not sure why, but I have to some
other things now.

@Ian with regards to pressure: Yeah, I just reinterpolated pressure again.
I guess at the time I didn't really look at the mantle transition zone
being messed up (see attached figure) and the center of the Earth missing.
It still frustrates me that I cannot reproduce it at greater depths, even
when I use a more detailed table. And I currently don't have pressure for
ak135 and iasp91 (but I could look around).

@Ian with regards to error message: I think this error is mostly useful to
allude people to the fact they are using the wrong unit. Maybe it should
say that?

On Tue, Jan 13, 2015 at 11:35 PM, Ian Rose <notifications@github.com> wrote:

> In burnman/seismic.py
> <https://github.com/geodynamics/burnman/pull/107#discussion-diff-22905349>
> :
>
> >
> >      def density(self, depth):
> >
> >          return self._lookup(depth, self.table_density)
> >
> >      def depth(self, pressure):
> > -        if pressure > max(self.table_pressure) or pressure < min(self.table_pressure)  :
> > -           raise ValueError, "Pressure outside range of PREM"
> > -        radius = np.interp(pressure, self.table_pressure[::-1], self.table_radius[::-1] )
> > -        return self.earth_radius - radius
> > +        if self.table_depth[0]:
>
> Should we no longer have this error check? According to the documentation
> for interp, it returns the value on the edge of the model points if you
> request a point outside of the interval.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/burnman/pull/107/files#r22905349>.
>